javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -162,7 +167,7 @@ private TreeMap(RedBlackTree<Tuple2<K, V>> entries) {
      */
     public static <K extends Comparable<? super K>, V> TreeMap<K, V> ofAll(java.util.Map<? extends K, ? extends V> map) {
         Objects.requireNonNull(map, "map is null");
-        RedBlackTree<Tuple2<K, V>> result = RedBlackTree.empty();
+        RedBlackTree<Tuple2<K, V>> result = RedBlackTree.empty(new EntryComparator<>(Comparator.naturalOrder()));
```

Comment:
@paplorinc, @ruslansennov We use the `naturalComparator()` only in the case we have no Comparable (generic) in scope. If we have a Comparable, like K here, we are able to make it Serializable like Ruslan showed. We use that solution in RedBlackTree for example.

So I would prefer to use

```java
final Comparator<Tuple2<K, V>> comparator = new EntryComparator<>((Comparator<? super K> & Serializable) K::compareTo);
RedBlackTree<Tuple2<K, V>> result = RedBlackTree.empty(comparator);
```

because the `naturalComparator()` is not Serializable.

We should write a simple Serialize/Deserialize test for the empty case here. There is a helper for that: `Serializables`, e.g.

```java
final CharSequence s = deserialize(serialize("test"));
assertThat(s).isEqualTo("test");
```